### ISSUE TITLE
Notebook does not work anymore (recent Ubuntu 22.04 LTS update ?)

### ISSUE THREAD

**Issue by BeauGeogeo** [googlecolab/colabtools#3863]
**Describe the current behavior**
My Colabd notebooks do not work as they used to anymore. Output of the exact same lines executed in the exact same order (cell order) is now different from what it used to be. The final output of the program is not working anymore, there is a bug with a library that I used to fix before and now I'm unable to find a workaround 

**Describe the expected behavior**
My Colabd notebooks working fine and as they used to until yesterday.

**What web browser you are using**
Chrome

**Additional context**
It is hard to tell how to reproduce the issue or giving a notebook reproducing it as the exact same notebook now produce a different output.
But the files I attached show differences in output.
Im wondering if this is linked to recent ubuntu update they talk about here : https://medium.com/google-colab/colab-updated-to-ubuntu-22-04-lts-709a91555b3c

Finally, I give a link to the version of my notebook before it was broken. Look at the output, especially for the commands installing python, or for the last function of the notebook where you can see the process has ended
correctly. Now if you re-run the notebook, you'll see that there's only one version of python you can choose, the output for the build of python is different, and new errors happen like missing packages 
Here's the link : https://colab.research.google.com/drive/1CHQhxocTtSegTpqyS6PGDWpqpVIiASuW?usp=sharing

Thank you very much and best regards,

Geoffrey

**Comment by mayankmalik-colab** [comment]
We do not have access to the notebook. Please give public access, so that we can have a look.

**Comment by BeauGeogeo** [comment]
> We do not have access to the notebook. Please give public access, so that we can have a look.

Yes sorry I thought the first link was correct. Here's the link with the permission for all users having the link I hope it's good this time : 

https://colab.research.google.com/drive/1CHQhxocTtSegTpqyS6PGDWpqpVIiASuW?usp=sharing

Thank you.

**Comment by metrizable** [comment]
@BeauGeogeo Thanks for providing the notebook. I see that in one of the cells you're installing and switching to Python 3.9 via update-alternatives. This is a scenario we do not support at this time. The Colab runtimes currently support Python 3.10.

**Comment by BeauGeogeo** [comment]
> @BeauGeogeo Thanks for providing the notebook. I see that in one of the cells you're installing and switching to Python 3.9 via update-alternatives. This is a scenario we do not support at this time. The Colab runtimes currently support Python 3.10.

Hello Sir and thank you for you reply. I'm sorry I do not totally get the scope of it yet. Because when I go to the fallback runtime of the environement, I have the choice between different versions of python, but not with the new ubuntu update apparently. 

But the main problem is that if you run the notebook with previous fallback runtime, no problem, but with the current version I have a bug I did not have before, when executing the last cell I have problems related to ninja library and c++ (which again I did not have with the fallback runtime where I can still execute everything fine).

Thank you again.
